java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageReadHolder.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.ipc.message;
+
+import java.nio.ByteBuffer;
+
+import org.apache.arrow.flatbuf.Message;
+
+/**
+ * Class to hold Message and message information when reading messages through MessageReader.
+ * The variables are defined as follows: messageLength is the length of the message read,
+ * messageBuffer contains the raw bytes of the message, message is the realized flatbuf Message.
+ */
+public class MessageReadHolder {
+  public int messageLength;
```

Comment:
The open struct interface is never the best. If we want to use it, we should clarify why in the class.